src/parser/ir/ir_parser.cc

```diff
@@ -98,34 +100,44 @@ const Operation& IrProgramParser::ConstructOperation(
                   const std::string& value_id =
                       CHECK_NOTNULL(named_value_context)->VALUE_ID()->getText();
                   // Operation Result Value
-                  auto find_value_id_operation = operation_map_.find(value_id);
-                  // check in place as we do not support operation lists yet.
-                  CHECK(find_value_id_operation == operation_map_.end());
-                  if (find_value_id_operation != operation_map_.end()) {
-                    return Value(value::OperationResult(
-                        *find_value_id_operation->second, value_id));
-                  }
-                  // Storage Value
-                  types::TypeFactory type_factory;
-                  return Value(value::StoredValue(
-                      Storage(value_id, type_factory.MakePrimitiveType())));
+                  auto find_value_id_operation = value_map_.find(value_id);
+                  CHECK(find_value_id_operation != value_map_.end()) << "Value must be found"; 
```

Comment:
Worth adding the missing value_id to the error message:
`"Value must be found: " << value_id`;

(You won't have to change the death test, I think.)

Also nit: "Value not found" might be a better message?